shortest_common_superstring/optimal_assignment.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+
+PRIMED = -1
+STARRED = -2
+
+def new_alternating_path(mtx, i, j):
+  pos = (i,j)
+  path = [pos]
+  while True:
+    starred_zeros = np.where(mtx.T[pos[1]] == STARRED)[0]
+    if len(starred_zeros) == 0:
+      break
+    pos = (starred_zeros[0], pos[1])
+    path.append(pos)
+    primed_zero = np.where(mtx[pos[0]] == PRIMED)[0][0]
+    pos = (pos[0], primed_zero)
+    path.append(pos)
+  for pos in path:
+    if mtx[pos] == STARRED:
+      mtx[pos] = 0
+    if mtx[pos] == PRIMED:
+      mtx[pos] = STARRED
+
+def cover_zeros(mtx):
+  marked_rows, marked_cols = (set(),set())
+
+  for i,row in enumerate(mtx):
+    for j,col in enumerate(mtx.T):
+      if mtx[i,j] == PRIMED:
+        mtx[i,j] = 0
+      if mtx[i,j] == STARRED:
```

Comment:
elif

Reply:
not quite, it is an if-elif block which should be independent from the first 'if' (some indices should satisfy first 'if' and then second 'elif')